### DISCUSSION TITLE
Traffic Selectors for GRE

### DISCUSSION THREAD

**Discussion by DavidBurns977** [strongswan/strongswan#1645] [General]
Hello,

I'm trying to set up a tunnel server, for the main campus end of a remote-campus tunnel.  (The remote side is an SRX.) 

With remote_ts = local_ts = dynamic[gre], the SA comes up, and both sides send ESP packets (of eg, pings or OSPF HELLOs), but neither responds to the other's ESP packets.  I'm including status information in this state.

With remote_ts = local_ts = 0.0.0.0/0, the SA comes up, pings and OSPF adjacencies work across the tunnel (as seen from the SRX), but the StrongSwan host loses all other connectivity, including my ssh -- so I can't get status information from that end.  Tearing down the tunnel from the SRX end restores connectivity on the StrongSwan host.

I've tried the docs, FAQ and made relevant searches, and would appreciate some help in figuring this out.

[config.txt](https://github.com/strongswan/strongswan/files/11155606/config.txt)
[srx-config.txt](https://github.com/strongswan/strongswan/files/11155607/srx-config.txt)
[status.txt](https://github.com/strongswan/strongswan/files/11155608/status.txt)
[charon.log](https://github.com/strongswan/strongswan/files/11155609/charon.log)

**Comment by tobiasbrunner** [comment]
> but neither responds to the other's ESP packets

What exactly do you mean? In the status output there is only inbound traffic, is that what you mean? Or can you also send outbound traffic but there is some routing/firewall issue that prevents an actual response to the decrypted traffic?

Note that you could avoid a bit of overhead by using transport mode instead of tunnel mode if the GRE tunnel is between the host's public IPs.

> With remote_ts = local_ts = 0.0.0.0/0, the SA comes up, pings and OSPF adjacencies work across the tunnel (as seen from the SRX), but the StrongSwan host loses all other connectivity, including my ssh -- so I can't get status information from that end.

Those traffic selectors obviously capture all traffic. You have to either install bypass policies to exclude certain traffic (e.g. [for SSH](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_for_specific_protocols_andor_ports)), use marks (to either manually mark traffic that should get tunneled or use a VTI interface to automatically do so via routing), or use XFRM interfaces to route traffic that should get tunneled.

**Reply by DavidBurns977** [reply]
Hello, and thank you for your reply.

I mean that, when I ping the strongswan host (10.208.211.10) from the SRX, tshark on enp1s0 shows ESP packets coming in to the strongswan host, but no ESP packets going back, and nothing from tshark on ipsec0. When I ping the SRX (10.208.211.9) from the strongswan host, tshark shows the ESP packets leaving (and ICMP outbound on ipsec0), but none coming back.

I don't think that the SRX supports transport mode for IKE-derived SAs.

When I selected GRE, I thought that it would be easy to do OSPF over, and that route table would select traffic for ipsec0 to be GRE encapsulated, and then dynamic[gre] would encrypt outgoing GRE, and accept incoming decrypted GRE.  I think I must be missing something about that?

**Reply by tobiasbrunner** [reply]
> I mean that, when I ping the strongswan host (10.208.211.10) from the SRX, tshark on enp1s0 shows ESP packets coming in to the strongswan host, but no ESP packets going back

OK, that's what the numbers in the status output show as well. What's the source address for that ping? Is it actually `10.208.211.9` or does the peer perhaps select a different IP?

> and nothing from tshark on ipsec0.

What do the stats for the interface say? (Check `ip -s tunnel show ipsec0`.)  You could maybe also use the iptables `TRACE` target to get more information about the decrypted packets, or maybe [`NFLOG`](https://docs.strongswan.org/docs/5.9/install/trafficDumps.html#_decapsulated_ipsec_traffic).

> When I ping the SRX (10.208.211.9) from the strongswan host, tshark shows the ESP packets leaving (and ICMP outbound on ipsec0), but none coming back.

Is there a way to check what's going on at the SRX? For instance, stats for the IPsec tunnel or GRE interface? Or even capture some traffic? It would be interesting to see if it is able to decrypt the packet and then maybe has an issue with the destination address (e.g. if it doesn't expect/allow traffic to `10.208.211.9`).

> I don't think that the SRX supports transport mode for IKE-derived SAs.

Weird, but OK.

> I think I must be missing something about that?

Not really. GRE isn't strictly necessary, though (of course depends on the other end, the advantage of GRE is that it's pretty portable). You can check the docs for other [route-based VPN](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html) options.
